scripts/qiita

```diff
@@ -29,7 +29,34 @@ from qiita_ware.ebi import EBISubmission
 from qiita_ware.commands import submit_EBI as _submit_EBI
 
 from re import search
+from threading import Thread
 
+import signal
+import sys
+
+watcher = None
+
+
+def watcher_sigint_handler(sig, frame):
+    print("Stopping Watcher...")
+    watcher.stop()
+    sys.exit(0)
+
+
+if qiita_config.plugin_launcher == 'qiita-plugin-launcher-qsub':
+    watcher = qdb.processing_job.Watcher()
+    watcher.start()
+    # wait for OK or QUIT
+    msg = watcher.queue.get(True)
+
+    if msg == 'QUIT':
+        print("Error: Watcher not running")
+        sys.exit(1)
+    else:
+        print("Watcher is running: %s" % msg)
```

Comment:
Sure, I'll add a message. The thing with multiprocessing.Process()es is that they're implicitly join()ed to their parent process by default. Hence w/out this handler, and with the Watcher running, you could no longer ctrl-c your way out of a running Qiita instance. The handler was added to make it behave as it did historically.

In general the Watcher process should be fine, because Process()es in the multiprocessing library are designed to behave like threads normally do. Anything that kills the parent process will kill the child processes as well.
 
That being said, I could see adding a handler for SIGTERM and possibly SIGQUIT, after reviewing the complete list of SIGnals in Linux. I think if someone wants to pause Qiita with a SIGSTOP, you'd still want the Watcher monitoring for completed jobs, as the metadata for a completed job doesn't remain in Torque/MOAB forever.

Since Qiita is still started and stopped in a screen session, I'd say we're good with just SIGINT for now. SIGTERM could be a new issue, IMHO.